type.go.tmpl

```diff
@@ -17,11 +17,11 @@
     {{else}}
       type: object
       {{- if gt (len $type.Fields) 0}}
+      {{- if gt (len $type.RequiredFields) 0}}
       required:
-      {{- range $i, $field := $type.Fields -}}
-        {{- if not $field.Optional }}
+      {{- range $i, $field := $type.RequiredFields }}
         - {{$field.Name}}
-        {{- end -}}
```

Comment:
{{- $required := false -}}

// in for loop
{{- $required = true -}}

Reply:
In the end I chose the other way: https://github.com/webrpc/webrpc/commit/8797f91fc1e5c6534b578f92842bea62fbc5a0ab